Apromore-Custom-Plugins/Process-Discoverer-Portal-Plugin/src/test/java/org/apromore/plugin/portal/processdiscoverer/common/ValidateFilenameTest.java

```diff
@@ -0,0 +1,89 @@
+/*-
+ * #%L
+ * This file is part of "Apromore Core".
+ * %%
+ * Copyright (C) 2018 - 2020 Apromore Pty Ltd.
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Lesser Public License for more details.
+ *
+ * You should have received a copy of the GNU General Lesser Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/lgpl-3.0.html>.
+ * #L%
+ */
+/**
+ * Test filename validation
+ *
+ * Implicitly test the regex used by zk validation
+ * in ApromoreCore/Apromore-Core-Components/Apromore-Portal/src/main/webapp/WEB-INF/ui.properties
+ */
+package org.apromore.plugin.portal.processdiscoverer.common;
+
+import static org.junit.Assert.fail;
+
+import org.apromore.plugin.portal.processdiscoverer.common.Constants;
+import org.junit.Test;
+
+import java.util.regex.Pattern;
+
+public class ValidateFilenameTest {
+
+    String[] VALID_FILENAMES = {
+            "Letters&0123456789",
+            "(Brackets) [Square Brackets]",
+            "hyphen-ated_underscored",
+            "filename+plus+plus",
+            "filename with spaces",
+            "汉字",
+            "\u9fff"
+    };
+
+    String[] INVALID_FILENAMES = {
+            "MoreThan60Chars012345678901234567890123456789012345678901234567890123456789",
+            "\r\n\t",
+            "",
+            "\u0040" // @ sign
+    };
+
+    @Test
+    public void testValidFilenames() {
+        String pattern = Constants.VALID_FILENAME_REGEX;
+
+        try {
+            for (int i = 0; i < VALID_FILENAMES.length; i++) {
+                String filename = VALID_FILENAMES[i];
+                if (!Pattern.matches(pattern, filename)) {
+                    fail("Invalid name " + filename);
+                }
+            }
+            System.out.println("Testing valid filenames");
+        } catch (Exception e) {
+            fail("Exception occurred: " + e.getMessage());
+        }
+    }
```

Comment:
I think this test aims to test whether PD will detect valid/invalid user input in the Input Dialog (org.apromore.plugin.portal.processdiscoverer.utils.InputDialog)? If so, can we relocate and rename it to do so making it easier to understand the purpose of this test? For example, it can be renamed as InputDialogTest, and the specific method to test here is InputDialog.okHandler, it can pass different values to txtValue, valuePatttern, etc. and check the expected value for labelError.

Reply:
If it is not straightforward to instantiate ZK UI components in the unit test, the InputDialog needs certain refactoring to do to separate the logic out of the UI code for testabbility, making the code easier to maintain as well. For example, add a new class called InputChecker under the same package with a checkInput method, this method has basically the same input arguments as InputDiaglog.okHandler but with String types rather than UI components. Then, create a unit test named InputCheckerTest with similar code as ValidateFilenameTest.

Reply:
Perhaps "ConstantsUnitTest" since it's just validating the correctness of the regexp VALID_FILENAME_REGEXP.

Reply:
Unit tests if written should reflect the context of the system being tested, and so the naming of the test classes should carry that meaning. In this case, the code to be tested is mostly related to the pattern (it should also test the missing input though), but the naming of the test class is quite important for somebody to read the code.

Reply:
I have moved the filename test to a more general portal.common since it may be needed by portal components and to separate logic and UI. The ZK related test is probably done better via e2e/selenium tests.

Reply:
Agreed, dreamind, just a minor comment: https://github.com/apromore/ApromoreCore/commit/8771d4e334d6f111f68e2037d3b7db44c6ae8a64#r43100604